dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java

```diff
@@ -34,12 +52,32 @@ public WorkflowVersion findByAlias(String alias) {
         return uniqueResult(namedTypedQuery("io.dockstore.webservice.core.WorkflowVersion.getByAlias").setParameter("alias", alias));
     }
 
-    public List<WorkflowVersion> getWorkflowVersionsByWorkflowId(long workflowId, int size, int firstResult) {
-        Query<WorkflowVersion> query = namedTypedQuery("io.dockstore.webservice.core.WorkflowVersion.getByWorkflowId");
-        query.setParameter("id", workflowId);
-        query.setFirstResult(firstResult);
-        query.setMaxResults(size);
-        return query.getResultList();
+    public List<WorkflowVersion> getWorkflowVersionsByWorkflowId(long workflowId, int limit, int offset, String sortOrder, String sortCol) {
+        CriteriaBuilder cb = currentSession().getCriteriaBuilder();
+        CriteriaQuery<WorkflowVersion> query = criteriaQuery();
+        Root<WorkflowVersion> version = query.from(WorkflowVersion.class);
+
+        List<Predicate> predicates = processQuery(sortCol, sortOrder, cb, query, version);
+        predicates.add(cb.equal(version.get("parent").get("id"), workflowId));
+        query.where(predicates.toArray(new Predicate[]{}));
+        query.select(version);
+
+        TypedQuery<WorkflowVersion> typedQuery = currentSession().createQuery(query).setFirstResult(offset).setMaxResults(limit);
+        return typedQuery.getResultList();
+    }
+
+    public List<WorkflowVersion> getPublicWorkflowVersionsByWorkflowId(long workflowId, int limit, int offset, String sortOrder, String sortCol) {
```

Comment:
If this method works the same as `getWorkflowVersionsByWorkflowId`, except for the exclusion of hidden versions, it might be worthwhile to combine the two and add an argument that controls whether that happens (or not).  Less duplicate code means less code to keep in sync...